pkg/ccl/changefeedccl/changefeed_stmt.go

```diff
@@ -384,18 +379,10 @@ func changefeedPlanHook(
 		}
 
 		// Start the job and wait for it to signal on startedCh.
-		errCh, err := sj.Start(ctx)
-		if err != nil {
+		if err := sj.Start(ctx); err != nil {
 			return err
 		}
-		select {
-		case <-ctx.Done():
-			return ctx.Err()
-		case err := <-errCh:
-			return err
-		case <-startedCh:
```

Comment:
so this is actually a change in behavior, right?

there's some code in `distChangefeedFlow` for which you dutifully make and pass a channel below, so that it can send on that ch in a little `finishedSetupFn` func that is passed to the distSQL planner to call after it starts up the flow. That func is sending on the ch, which prior to this change, was the "resultsCh" passed to the job, ie. startedCh above, passed in the usual resultCh position.


With that, then this recv would _block_ until the flow was setup (or one of the errors above), or until an error popped out of errCh. Without a call to await completion, we no longer see those execution errors that occur after creation but before startedCh would have unblocked, and without startedCh, we now return the job ID immediately, even if it will fail to start. Maybe that's okay, but it seems like there was intent here to avoid that.

We could try to use the ReportResults API instead here: put a ch on the changefeed resumer, have the `finishedSetupFn` callback func close it (along with a deferred close, in case it never makes it there), and then have `ReportResults` recv on it and then return nil (can ignore its resultsCh arg). Then a call to ReportResults right here would get us that same block-until-the-flow-is-up semantics. 

However we still aren't getting back exec errors, whereas I think we were if they occured before the flow it started and closed/sent on startedCh before. That'd be a little tricker, but I think we could in theory have Resume see if an error popped out of distflow, if so, store it in the resumer and then when we unblock on startedCh, we could return contents of that. Does that all make sense?